htlcswitch/link.go

```diff
@@ -2151,15 +2154,15 @@ func (l *channelLink) processRemoteAdds(fwdPkg *channeldb.FwdPkg,
 			err = l.channel.SettleHTLC(preimage,
 				pd.HtlcIndex, pd.SourceRef, nil, nil)
 			if err != nil {
-				l.fail("unable to settle htlc: %v", err)
+				l.fail(true, "unable to settle htlc: %v", err)
 				return false
 			}
 
 			// Notify the invoiceRegistry of the invoices we just
 			// settled with this latest commitment update.
 			err = l.cfg.Registry.SettleInvoice(invoiceHash)
 			if err != nil {
-				l.fail("unable to settle invoice: %v", err)
+				l.fail(true, "unable to settle invoice: %v", err)
```

Comment:
This might be a bit too aggressive as this may be an issue with our db state or the implementation of the interface.